### ISSUE TITLE
False positive for yoda_test_linter in magrittr pipeline

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1066]
A pattern seen a lot in `formula.tools`:

https://github.com/decisionpatterns/formula.tools/blob/45b6654e4d8570cbaf1e2fd527652471202d97ad/tests/testthat/test-lhs.r#L13

```
lhs( NA ~ . )   %>% expect_equal(NA) 
```

IMO this is not the best way to write tests, but nevertheless we shouldn't trigger a lint with `yoda_test_linter()`.

**Comment by MichaelChirico** [comment]
closed by #1065
